### PR TITLE
feat(create-collector-step1): apply collector api & infinite scroll

### PR DESCRIPTION
**Pull request by sulmoJ**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Things to Talk About
SSIA

https://github.com/cloudforet-io/console/assets/50662170/c09e39db-16c9-4954-8e55-d5bba0690c3d

### CODE REVIEW

File: apps/web/src/services/asset-inventory/collector/create-collector/modules/Step1SearchFilter.vue [hunk 1]
```diff
@@ -114,15 +125,44 @@ const state = reactive({
...
+    }
+};
+
+const handleChangeProvider = (provider) => {
+    console.log('provider', provider);
+    collectorPageStore.setSelectedProvider(provider);
+};
+
+watch(() => state.selectedRepository, (repository) => {
+    emit('selectedRepository', repository);
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/create-collector/modules/Step1SearchFilter.vue:158]: nitpick: I think this should be `select-repository`

File: apps/web/src/services/asset-inventory/collector/create-collector/modules/CreateCollectorStep1.vue [hunk 2]
```diff
@@ -80,47 +85,54 @@ const state = reactive({
...
-    selectedProvider: '',
+    currentPage: 1,
+    totalCount: 0,
 });
+const pluginCardListRef = ref<HTMLElement | null>(null);
+
 
 const pluginApiQuery = new ApiQueryHelper();
-const getPlugins = async () => {
+const getPlugins = async (): Promise<CollectorPluginModel[]> => {
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/create-collector/modules/CreateCollectorStep1.vue:95]: praise: 👍

File: apps/web/src/services/asset-inventory/collector/create-collector/modules/CreateCollectorStep1.vue [hunk 3]
```diff
@@ -80,47 +85,54 @@ const state = reactive({
...
 
-const handleSearch = (value) => {
+const loadMorePlugin = async () => {
+    if (state.totalCount <= state.pluginList.length) {
+        return;
+    }
+    state.currentPage += 1;
+    state.loading = true;
+    const additionalPlugin = await getPlugins();
+    state.pluginList = [...state.pluginList, ...additionalPlugin];
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/create-collector/modules/CreateCollectorStep1.vue:130]: suggestion: How about to use `concat()`? Just suggestion.

**Review by yuda110 — APPROVED** [review]
LGTM! Considered conventions more.

**Review by skdud4659 — APPROVED** [review]
LGTM!!

**Pull request merged**